app/test4.py

```diff
@@ -0,0 +1,33 @@
+import git
+from git.exc import GitCommandError
+
+def push_changes(repo_path, commit_message, branch='main'):
+    """
+    Push changes to a remote git repository.
+
+    Args:
+    repo_path (str): Path to the local git repository.
+    commit_message (str): Commit message for the changes.
+    branch (str): Branch to which the changes should be pushed. Defaults to 'main'.
+
+    Returns:
+    str: Success message or error message.
+    """
+    try:
+        repo = git.Repo(repo_path)
+        if not repo.is_dirty():
+            return "No changes detected in the repository."
+        repo.git.add(A=True)
+        repo.index.commit(commit_message)
+        origin = repo.remote(name='origin')
+        origin.push(refspec=f'{branch}:{branch}')
+        return "Changes pushed successfully to the remote repository."
+    except GitCommandError as e:
+        return f"An error occurred while pushing changes: {str(e)}"
+    except Exception as e:
```

Comment:
Catch more specific exceptions before the general `Exception` to handle known errors more gracefully and provide more detailed error messages for those cases.

**Label**: best practice

**Code suggestion**

```python
except GitCommandError as e:
    return f"An error occurred while pushing changes: {str(e)}"
except SomeSpecificError as e:
    return f"A specific error occurred: {str(e)}"
except Exception as e:
```